### PR TITLE
Add OddDotNet registry files

### PR DESCRIPTION
**Pull request by DoubleTK**
I'm requesting to add OddDotNet and the CSharp client to the registry.

### CODE REVIEW

**Comment by svrnm** [comment]
@DoubleTK do not worry about the CI checks failing, we can help you with that for merging the PR

File: data/registry/otel-odddotnet.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  OddDotNet enables automated Observability Driven Development, hence where it
+  gets its name.
+authors:
+  - name: Tyler Kenna
+    email: odddotnet@gmail.com
+    url: https://github.com/DoubleTK
+createdAt: '2024-11-08'
+isNative: true
+isFirstParty: false
```
Comment (svrnm) [inline data/registry/otel-odddotnet.yml:31]:
```suggestion
```

remove that, those flags are not applicable for utilities

Comment (DoubleTK) [reply data/registry/otel-odddotnet.yml]: Done 👍

File: data/registry/otel-odddotnet.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# cSpell:ignore Kenna odddotnet
+title: OddDotNet OTel Collector Test Harness
```
Comment (svrnm) [inline data/registry/otel-odddotnet.yml:2]:
```suggestion
title: OddDotNet: Test Harness for OpenTelemetry Collector
```

File: data/registry/otel-odddotnet.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  - test-harness
+  - odddotnet
+urls:
+  repo: https://github.com/OddDotNet/OddDotNet
+  docs: https://odddotnet.github.io/OddDotDocs/
+license: Apache 2.0
+description: OddDotNet is an OpenTelemetry Test Harness that works for any
+  language, built in .NET. It is a container image that accepts OpenTelemetry
+  signals over gRPC (and soon HTTP), and it includes a query language for
+  verifying and validating those signals.
```
Comment (svrnm) [inline data/registry/otel-odddotnet.yml:21]:
```suggestion
description: OddDotNet is a Test Harness for OpenTelemetry that works for any
  language, built in .NET. It is a container image that accepts OpenTelemetry
  signals over gRPC, and it includes a query language for
  verifying and validating those signals.
```

Let's not make forward looking statements (and soon HTTP) in the registry entries. You can come back and update the description when the HTTP support is availab.e

File: data/registry/tools-odddotnet-client-dotnet.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  - observability
+  - test-harness
+urls:
+  repo: https://github.com/OddDotNet/OddDotCSharp
+  docs: https://odddotnet.github.io/OddDotDocs/
+license: Apache 2.0
+description: OddDotNet is an OpenTelemetry Test Harness that works for any
+  language, built in .NET. It is a container image that accepts OpenTelemetry
+  signals over gRPC (and soon HTTP), and it includes a query language for
+  verifying and validating those signals.
```
Comment (svrnm) [inline data/registry/tools-odddotnet-client-dotnet.yml:20]:
```suggestion
description: OddDotNet is an Test Harness for OpenTelemetry that works for any
  language, built in .NET. It is a container image that accepts OpenTelemetry
  signals over gRPC, and it includes a query language for
  verifying and validating those signals.
```

**Review by svrnm — CHANGES_REQUESTED** [review]
a few more comments

File: data/registry/otel-odddotnet.yml (hunk 2 above)
Comment (DoubleTK) [reply data/registry/otel-odddotnet.yml:2]: I like this change, but now it has me wondering if I should remove "Collector" altogether. It's really a test harness for OpenTelemetry in general, so "OddDotNet: Test Harness for OpenTelemetry" might make more sense. I'll make that change, and if that doesn't work for you I'll change it to what you suggested above tomorrow. Thanks!

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
@DoubleTK can you update the PR to be in sync with the base branch?

Note: Creating PRs from an ORG account like https://github.com/OddDotNet/ does not allow maintainers to edit your PR, so a lot of our automation does not work. Therefor for future PRs I recommend you fork from your user handle (DoubleTK) and push your PRs from there.

**Comment by DoubleTK** [comment]
> @DoubleTK can you update the PR to be in sync with the base branch?
> 
> Note: Creating PRs from an ORG account like https://github.com/OddDotNet/ does not allow maintainers to edit your PR, so a lot of our automation does not work. Therefor for future PRs I recommend you fork from your user handle (DoubleTK) and push your PRs from there.

Ah, that's good to know, I wasn't aware of that limitation. Thank you for the info, I'll make sure to do that next time. Updating branch with latest main now.

**Comment by svrnm** [comment]
@DoubleTK can you run `npm run fix:all` locally and update the changes?

**Comment by svrnm** [comment]
@DoubleTK the issue is that you need to have your URLs added to the refcache. Since we can not edit this PR for you, there are 2 options:

* run `npm run fix:refcache` locally and commit the changes
* make this PR editable, [following these instructions](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) or re-raise the PR using a non-org handle

**Comment by DoubleTK** [comment]
> @DoubleTK the issue is that you need to have your URLs added to the refcache. Since we can not edit this PR for you, there are 2 options:
> 
> * run `npm run fix:refcache` locally and commit the changes
> * make this PR editable, [following these instructions](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) or re-raise the PR using a non-org handle

I think I'm going to take that second option and redo this using my personal account rather than org. I'll get that PR created first, and then paste the link here and abandon this PR.

**Comment by DoubleTK** [comment]
Okay I've created a new PR using my personal to allow for maintainer edits. Here's the link:

https://github.com/open-telemetry/opentelemetry.io/pull/5617.

Thank you again for your help and patience, I appreciate it.

**Pull request closed without merging**